core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/debug/DebuggerManager.java

```diff
@@ -26,36 +28,111 @@
  * @author Anatoliy Bazko
  */
 @Singleton
-public class DebuggerManager {
-    private final AppContext appContext;
-
-    private Map<String, Debugger> debuggers;
-    private Debugger debugger;
+public class DebuggerManager implements DebuggerManagerObservable, DebuggerObserver {
```

Comment:
Maybe it would be better to extract and interface from this manager? It will help you to write less code to cover with tests. wdyt?